enhancements/microshift/kubernetes-for-device-edge.md

```diff
@@ -0,0 +1,515 @@
+   network to upgrade devices already in the field using minimal
+   network bandwidth.
+5. As a device administrator, I can specify application-specific logic
+   with which the device will verify a successful upgrade and roll
+   back the upgrade automatically if verification fails.
+6. As a device administrator, I have control over the operating
+   system running under MicroShift.
+
+### Goals
+
+* Describe an approach to providing a version of Kubernetes compatible
+  with the OpenShift APIs that it includes and that meets the
+  constraints described in the Motivation section.
+* Identify the minimal set of APIs that will be included and the
+  components needed to support them.
+* Design a system that can work autonomously, without requiring
+  external orchestration tools for deployment.
+* Treat MicroShift as an application running on RHEL, with the minimum
+  privileges possible.
+* Design a system that can be configured by someone familiar with
+  Linux, without Kubernetes expertise.
+* Follow the [MicroShift design
+  principles](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+### Non-Goals
+
+* We are not trying to fit all of the features of a standalone
+  OpenShift deployment into the target hardware platforms.
+* We are not trying to make MicroShift self-managing, so no core
+  OpenShift operators will be included.
+* We do not plan to support multi-node clusters.
+* The details of the default CSI driver are left to a future
+  enhancement.
+* The details of the default CNI driver are left to a future
+  enhancement.
+* Use cases beyond industrial applications, such as developer
+  environments, are out of scope for now.
+* The design of MicroShift relies heavily on features of RHEL for Edge
+  and rpm-ostree. Those features are referenced here, but their
+  implementation is not described in detail.
+* The details of integrating with fleet orchestration tools such as
+  MCE or ACM are left to a future enhancement.
+* We do not want to push MicroShift use on RHEL versions where OCP is
+  not supported. We do have interest from users in RHEL 9, but will
+  not GA on RHEL 9 faster than OCP does.
+* We do not want to push MicroShift use on hardware platforms where
+  RHEL for Edge is not supported. We may use interest in MicroShift to
+  drive prioritization of hardware support, but we would not support
+  MicroShift on a platform where RHEL is not supported.
+
+## Proposal
+
+To meet the resource consumption constraints (RAM and CPU), MicroShift
+is designed as an all-in-one binary for the components that are
+required to launch a Kubernetes control plane, combined with
+components running in containers to provide runtime services after the
+control plane is up.
+
+Today the all-in-one binary includes:
+
+* etcd
+* Kubernetes API server
+* Kubernetes controller manager
+* Kubernetes scheduler
+* Kubelet
+* Kube Proxy Server
+* OpenShift API server
+* OpenShift controller manager
+* OpenShift SCC manager
+
```

Comment:
In Openshift, etcd is not encrypted.
https://docs.openshift.com/container-platform/4.10/security/encrypting-etcd.html
With the all-in-one-binary, is it also unencrypted with the option of encrypting?

Reply:
I think the plan is to rely on RHEL's full disk encryption, right @fzdarsky?

Reply:
Thanks for clarifying this @fzdarsky @dhellmann  👍